test-clients/src/main/java/com/hedera/services/bdd/suites/ethereum/EthereumSuite.java

```diff
@@ -52,26 +48,189 @@ public class EthereumSuite extends HapiApiSuite {
+								.via("autoAccount"),
+						getTxnRecord("autoAccount").andAllChildRecords(),
+						uploadInitCode(PAY_RECEIVABLE_CONTRACT),
+						contractCreate(PAY_RECEIVABLE_CONTRACT).adminKey(THRESHOLD)
+				).when(
+						// Network and Node fees in the schedule for Ethereum transactions are 0,
+						// so everything charged will be from the gas consumed in the EVM execution
+						uploadDefaultFeeSchedules(GENESIS)
+				).then(
+						withOpContext((spec, ignore) -> {
+							final String senderBalance = "senderBalance";
+							final String payerBalance = "payerBalance";
+							final var subop1 =
+									balanceSnapshot(senderBalance, SECP_256K1_SOURCE_KEY)
+											.accountIsAlias();
+							final var subop2 = balanceSnapshot(payerBalance, RELAYER);
+							final var subop3 = ethereumCall(PAY_RECEIVABLE_CONTRACT, "deposit", depositAmount)
+									.type(EthTxData.EthTransactionType.EIP1559)
+									.signingWith(SECP_256K1_SOURCE_KEY)
+									.payingWith(RELAYER)
+									.via("payTxn")
+									.nonce(0)
+									.maxGasAllowance(fullAllowance)
+									.maxFeePerGas(partialFee)
+									.gasLimit(gasLimit)
+									.sending(depositAmount)
+									.hasKnownStatus(ResponseCodeEnum.SUCCESS);
+
+							final HapiGetTxnRecord hapiGetTxnRecord = getTxnRecord("payTxn").logged();
+							allRunFor(spec, subop1, subop2, subop3, hapiGetTxnRecord);
+
+							var fees = hapiGetTxnRecord.getResponseRecord().getTransactionFee();
+							final var subop4 = getAliasedAccountBalance(SECP_256K1_SOURCE_KEY).hasTinyBars(
+									changeFromSnapshot(senderBalance, -depositAmount-senderCharged));
+							final var subop5 = getAccountBalance(RELAYER).hasTinyBars(
+									changeFromSnapshot(payerBalance, -(fees-senderCharged)));
+							allRunFor(spec, subop4, subop5);
+						})
+				);
+	}
+
+
+	List<HapiApiSpec> feePaymentMatrix() {
+		final long gasPrice = 47;
+
+		final long noPayment = 0L;
+		final long partialFee = gasPrice / 3;
+		final long partialPayment = partialFee * gasLimit;
+		final long fullAllowance = gasPrice * gasLimit * 5/4;
+
+		return Stream.concat(Stream.of(bothPayFeesSucceeds()), Stream.of(
+				new Object[] { false, noPayment, noPayment, false, false},
+				new Object[] { false, noPayment, partialPayment, false, false },
+				new Object[] { true, noPayment, fullAllowance, false, true },
+				new Object[] { false, partialFee, noPayment, false, false },
+				new Object[] { false, partialFee, partialPayment, false, false },
+//				new Object[] { true, partialFee, fullAllowance, false, false }, // done in bothPayFeesSucceeds()
+				new Object[] { true, gasPrice, noPayment, true, false },
+				new Object[] { true, gasPrice, partialPayment, true, false },
+				new Object[] { true, gasPrice, fullAllowance, true, false}
+		).map(params ->
+				// [0] - success
+				// [1] - sender gas price
+				// [2] - relayer offered
+				// [3] - is sender charged
```

Comment:
_if_ they are charged is insufficient.  (a) we need to validate _how much_ especially in the mixed payment scenario and (b) this isn't asserting that zero fees are charged when set to zero